scmailclient/scmail.py

```diff
@@ -0,0 +1,493 @@
+            logging.StreamHandler(),
+        ],
+    )
+    ctx.logger = logging.getLogger(__name__)
+
+    # Set config file path.
+    ctx.config = Path("scmail.conf")
+
+    # If the config not exist, set user information. If not, run initial gpg.
+    gnupghome = (
+        load_config()
+        if ctx.config.exists()
+        else ctx.invoke(
+            set_user_info,
+            gnupghome=click.prompt("Enter your gnupg home dir:", default="gnupgkeys"),
+        )
+    )
+    ctx.logger.debug(f"gnupg home is {gnupghome}")
+
+    ctx.gpg = GpgOpt(mygnupghome=gnupghome)
+
+    # Initial pretty printer
+    ctx.pp = pprint.PrettyPrinter(indent=2)
+
+    logging.info("Initialize client successful.")
+
+
+@click.pass_context
+def load_config(ctx):
+    """Load config file."""
+    f = ctx.config.open("r").read()
+    config = re.split("\n|=", f)
+    logging.debug(f"read config file:{config}")
+
+    for i in range(0, len(config) - 1, 2):
+        print(f"The {config[i]} is: {config[i+1]}")
+
+    print()
+    logging.info("Config file exists. Load Successful.")
+    return config[1]
+
+
+@click.pass_context
+def set_user_info(ctx, gnupghome):
+    """Set user information and return gnupghome and email."""
+    config = "gnupghome=" + gnupghome + "\n"
+
+    # Store config file and other information.
+    ctx.config.touch(exist_ok=False)
+    with ctx.config.open("w") as f:
+        f.write(config)
+
+    logging.info("Setting user information finish.")
+    return gnupghome
+
+
+@click.command(name="create")
+@click.option(
+    "--name",
+    "-n",
+    prompt="Enter Name",
+    default=getpass.getuser(),
+    type=str,
+    help="Generator name.",
+)
+@click.option(
+    "--email",
+    "-e",
+    prompt="Enter Email",
+    default=f"{getpass.getuser()}@scmail.dev",
+    type=str,
+    help="The user email address.",
+)
+@click.option(
+    "--key-type",
+    "-t",
+    prompt="Enter key type",
+    default="RSA",
+    type=str,
+    help="The algorithms to generate the key. ex. RSA",
+)
+@click.option(
+    "--key-length",
+    "-l",
+    prompt="Enter key length",
+    default=1024,
+    type=int,
+    help="The length of the key.",
+)
+@click.option(
+    "--expire-date",
+    "-d",
+    prompt="Enter expire date",
+    default="2y",
+    type=str,
+    help="The expire time of the key pair. 0 or empty for never expire.",
+)
+@click.password_option(
+    "--password",
+    "-p",
+    prompt="Enter private key password",
+    type=str,
+    help="Password of private key.",
+)
+@click.pass_context
+def create_key(ctx, name, email, key_type, key_length, expire_date, password):
+    """Create gnupg key pairs."""
+    logging.debug(
+        f"Information about key:\nName: {name}\nEmail: {email}\nkey type: {key_type}\nkey length: {key_length}\nexpire date: {expire_date}\n"
+    )
+
+    # Warning if key never expire and user want to continue.
+    if expire_date == 0 and not click.confirm(
```

Comment:
Should this be `... and click.confirm(...)`? I'm not familiar with `.confirm()` but this logic seems backwards.

Reply:
Thanks, I changed this sentence and add an `exit` if confirm not.